python/ql/test/experimental/library-tests/CallGraph/CallGraphTest.qll

```diff
@@ -0,0 +1,145 @@
+import python
+
+/** Gets the comment on the line above for a given `ast_node` */
+Comment comment_for(AstNode ast_node) {
+    exists(int line | line = ast_node.getLocation().getStartLine() - 1 |
+        result
+                .getLocation()
+                .hasLocationInfo(ast_node.getLocation().getFile().getAbsolutePath(), line, _, line, _)
+    )
+}
+
+/** Gets the value from `tag:value` in the comment for `ast_node` */
+string getAnnotation(AstNode ast_node, string tag) {
+    exists(Comment comment, string match, string the_regex |
+        the_regex = "([\\w]+):([\\w.]+)" and
+        comment = comment_for(ast_node) and
+        match = comment.getText().regexpFind(the_regex, _, _) and
+        tag = match.regexpCapture(the_regex, 1) and
+        result = match.regexpCapture(the_regex, 2)
+    )
+}
+
+/** Gets a callable annotated with `name:name` */
+Function annotatedCallable(string name) { name = getAnnotation(result, "name") }
+
+/** Gets a call annotated with `calls:name` */
+Call annotatedCall(string name) { name = getAnnotation(result, "calls") }
+
+predicate missingAnnotationForCallable(string name, Call call) {
+    call = annotatedCall(name) and
+    not exists(annotatedCallable(name))
+}
+
+predicate nonUniqueAnnotationForCallable(string name, Function callable) {
+    strictcount(annotatedCallable(name)) > 1 and
+    callable = annotatedCallable(name)
+}
+
+predicate missingAnnotationForCall(string name, Function callable) {
+    not exists(annotatedCall(name)) and
+    callable = annotatedCallable(name)
+}
+
+/** There is an obvious problem with the annotation `name` */
+predicate name_in_error_state(string name) {
```

Comment:
nah, that's just me being inconsistent. I have transformed everything into camelCase now. I guess some of them could be private, but seeing as they're in a path that can't be imported from anywhere (since it contains `-`), I don't really see the point in thinking too much about this.